airflow/providers/databricks/triggers/databricks.py

```diff
@@ -84,13 +84,30 @@ async def run(self):
         async with self.hook:
             while True:
                 run_state = await self.hook.a_get_run_state(self.run_id)
+                notebook_error = None
                 if run_state.is_terminal:
+                    if run_state.result_state == "FAILED":
+                        run_info = await self.hook.a_get_run(self.run_id)
+                        task_run_id = None
+                        if "tasks" in run_info:
+                            for task in run_info["tasks"]:
+                                if task.get("state", {}).get("result_state", "") == "FAILED":
+                                    task_run_id = task["run_id"]
+                        if task_run_id is not None:
```

Comment:
If more than one task failed in the tasks, this `task_run_id` will keep getting overwritten while iterating in the earlier for loop. Should we not collect error messages for all the tasks that failed?

Reply:
ideally yes, but this logic was there for `RunNowOperator` before, so to maintain same error, i just cut copied with async calling

Reply:
let me know your thoughts on it, will change and append all task error in error message.

Reply:
Also good to check if the task["run_id"] different for all the tasks in a run or it's the same?

Reply:
task["run_id"] is different for all the tasks, should i keep a map of task run id and error message? @pankajkoti

Reply:
https://github.com/apache/airflow/pull/39110#discussion_r1579863752

Reply:
regarding to this, I'm thinking of keeping it as it is now (not collecting the error message) for keeping the behavior consistent. We can create another issue and PR for fixing both non-deferrable and deferrable all along. WDYT?

Reply:
Adding the non-deferrable part of it to this PR might be a bit huge. That why I'm proposing another PR for this fix

Reply:
Same, i can work with different pr for it, so @Lee-W @pankajkoti  are we merging this pr?  once merge will take pull from master and will work on @pankajkoti  idea to collect all errors

Reply:
Wasn't @Lee-W suggesting to extract only the non-deferrable part into a separate PR? I believe including the correct version for the deferrable part is still appropriate for this PR, WDYT?

Reply:
We can term that the non-deferrable part is buggy. But since that is buggy, I don't think it makes sense to add the new code as incomplete/incorrect just to keep consistency.

Reply:
After a second thought, @pankajkoti 's suggestion might be better. @gaurav7261 Sorry for the misleading suggestion previously 😞  Let's work on fixing the deferrable part first and non-deferrable part in another PR. Thanks!

Reply:
ok working in same this pr,  thanks